app/controller.go

```diff
@@ -93,19 +93,23 @@ func (fn HTTPHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 func (c *Controller) GetDeviceFiles(w http.ResponseWriter, r *http.Request, user UserInfo) error {
+	zone := mux.Vars(r)["zone"]
+	host := mux.Vars(r)["host"]
```

Comment:
It's not a good idea to repeat these two lines too many times, especially since "zone" and "host" are used as literals. Could you encapsulate extracting the host and zone parameters from the request into a utility function?

Reply:
Done, created to independent utility functions.